File: solutions/java/two-fer/3/src/main/java/Twofer.java
import java.util.Optional;

class Twofer {
    static String twofer(String name) {
        // Java does not support default parameters
        // String you = name != null ? name : "you";

        // Using java.util.Optional
        String you = Optional.ofNullable(name).orElse("you");
        
        return "One for " + you + ", one for me.";
    }
}

File: solutions/c/two-fer/3/src/two_fer.c
#include "two_fer.h"
#include <string.h>
#include <stdio.h>

#define BUFFER_SIZE (100)

void two_fer(char *buffer, const char *name)
{
    snprintf(buffer, BUFFER_SIZE, "One for %s, one for me.", name == NULL ? "you" : name);
}